pkg/crc/services/dns/dns_darwin.go

```diff
@@ -144,35 +146,14 @@ func waitForNetwork() error {
 // updateHostsConfFile updates the host's /etc/hosts file with Instance IP.
 func updateHostsConfFile(instanceIP string, hostnames ...string) error {
 	// Get the current value of /etc/hosts file
-	hosts, err := goodhosts.NewHosts()
-	if err != nil {
-		return err
-	}
+	goodhostPath := filepath.Join(constants.CrcBinDir, constants.GoodhostBinaryName)
 	for _, hostname := range hostnames {
-		if err := replaceHostname(&hosts, instanceIP, hostname); err != nil {
+		if _, _, err := crcos.RunWithDefaultLocale(goodhostPath, "rm", hostname); err != nil {
 			return err
 		}
-	}
-
-	// Flush operation is required just after replacing the host this
-	// add the entry to /etc/hosts file
-	if err := hosts.Flush(); err != nil {
-		return err
-	}
-	return nil
-}
-
-func replaceHostname(hosts *goodhosts.Hosts, ipAddress string, hostname string) error {
-	logging.Debugf("Updating %s to %s in /etc/hosts file", hostname, ipAddress)
-	if hosts.HasHostname(hostname) {
-		if err := hosts.RemoveByHostname(hostname); err != nil {
+		if _, _, err := crcos.RunWithDefaultLocale(goodhostPath, "add", instanceIP, hostname); err != nil {
 			return err
```

Comment:
Would probably be nicer (more readable) to have a helper to add/remove an entry to /etc/hosts.